test/e2e/create_test.go

```diff
@@ -107,14 +107,27 @@ func (tc *testContext) deleteMachineSet(ms *mapi.MachineSet) error {
 }
 
 // waitForWindowsMachines waits for a certain amount of Windows Machines to reach a certain phase
+// if machineCount = 0, it implies we are only waiting for Machines to be deleted and the phase is
+// ignored in this case.
 func (tc *testContext) waitForWindowsMachines(machineCount int, phase string) error {
-	machineCreationTimeLimit := time.Minute * 5
+	if machineCount == 0 && phase != "" {
+		return errors.New("expected phase to be to be an empty string if machineCount is 0")
+	}
+
+	var machines *mapi.MachineList
```

Comment:
This won't be needed if you don't reuse the `err` variable below.

Reply:
this is needed to print the machines.Items in the log at the end of the function as the machines variable declared in poll function has a limited scope.

Reply:
Shouldn't you be using `machineCount` in the print statement?

Reply:
I think that will give an inaccurate time log, consider the case where it timed out on reaching the desired number of Machines. In this case the log will print the time with the original machineCount(desired) passed by the caller instead of the current number of Machines that took the time printed.

Reply:
But what we care about is reaching the desired number of Machines and not the current number of Machines.

Reply:
I agree with your statement, that is what we care about. My only concern here is the accuracy of the log, the time printed in the log statement would not correspond to the time it took to create those Machines. 
Going by the statement "%v time is required for %v Machines to reach phase %s" , if a timeout occurs and only n machines are created/deleted by that time we will get an accurate indication of how much time was taken to do that. It might not be a huge concern in this case but just a thought.

Reply:
But won't the [log](https://github.com/openshift/windows-machine-config-operator/blob/master/test/e2e/create_test.go#L128) within the loop clarify this?

Reply:
Anyhow we are just bikeshedding now 😄 I am fine with whatever approach you decide on.